NWNXLib/Utils.cpp

```diff
@@ -271,5 +272,31 @@ bool CompareVariables(API::CNWSScriptVarTable *pVars1, API::CNWSScriptVarTable *
     return true;
 }
 
+API::CNWSScriptVarTable *GetScriptVarTable(API::CGameObject *pObject)
+{
+    if (!pObject)
+        return nullptr;
+
+    switch (pObject->m_nObjectType)
+    {
+        case OBJECT_TYPE_AREA:
+            return &static_cast<API::CNWSArea*>(pObject)->m_ScriptVars;
+        case OBJECT_TYPE_MODULE:
+            return &static_cast<API::CNWSModule*>(pObject)->m_ScriptVars;
+        default:
+            return &static_cast<API::CNWSObject*>(pObject)->m_ScriptVars;
+    }
+}
+
+void DestroyGameEffect(API::CGameEffect* pEffect)
+{
+    if (pEffect)
+    {
+        // Don't actually need the this pointer here.
+        static_cast<API::CNWVirtualMachineCommands*>(nullptr)->DestroyGameDefinedStructure(0, pEffect);
```

Comment:
Maybe it would be better to get the VM rather than casting nullptr? I think this is undefined behaviour, and even if it wasn't, it might cause issues later down the line when niv changes that function to spam cats or something.

Reply:
This isn't just `CVirtualMachine`, it's some other singleton that is surprisingly hard to get at. But I'll try.